traffic_control/models.py

```diff
@@ -0,0 +1,118 @@
+import uuid
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.contrib.gis.db import models
+from django.utils.translation import ugettext_lazy as _  # NOQA
+
+
+class TrafficSignCode(models.Model):
+    id = models.UUIDField(
+        primary_key=True, unique=True, editable=False, default=uuid.uuid4
+    )
+    code = models.CharField(_("Code"), max_length=32)
+    description = models.CharField(
+        _("Description"), max_length=254, blank=True, null=True
+    )
+
+    def __str__(self):
+        return "%s - %s" % (self.code, self.description)
+
+
+class Lifecycle(models.Model):
+    id = models.UUIDField(
+        primary_key=True, unique=True, editable=False, default=uuid.uuid4
+    )
+    status = models.CharField(_("Status"), max_length=32)
+    description = models.CharField(
+        _("Description"), max_length=254, blank=True, null=True
+    )
+
+    def __str__(self):
+        return "%s" % self.description
+
+
+class Size(models.TextChoices):
+    SMALL = "S", _("Small")
+    MEDIUM = "M", _("Medium")
+    LARGE = "L", _("Large")
+
+
+class Surface(models.TextChoices):
+    CONVEX = "CV", _("Convex")
+    DIRECT = "DR", _("Direct")
+
+
+class Color(models.TextChoices):
+    BLUE = "BL", _("Blue")
+    YELLOW = "YE", _("Yellow")
+
+
+class TrafficSignPlan(models.Model):
+    id = models.UUIDField(
+        primary_key=True, unique=True, editable=False, default=uuid.uuid4
+    )
+    location_xy = models.PointField(_("Location (2D)"), srid=settings.SRID)
+    location_z = models.DecimalField(
+        _("Location Z"), max_digits=5, decimal_places=2, blank=True, null=True
+    )
+    height = models.DecimalField(
+        _("Height"), max_digits=5, decimal_places=2, blank=True, null=True
+    )
+    direction = models.DecimalField(
+        _("Direction"), max_digits=5, decimal_places=2, blank=True, null=True
+    )
+    parent = models.ForeignKey("self", on_delete=models.CASCADE, blank=True, null=True)
+    decision_date = models.DateField(_("Decision date"))
+    code = models.ForeignKey("TrafficSignCode", on_delete=models.CASCADE)
+    value = models.CharField(
+        _("TrafficSignCode value"), max_length=32, blank=True, null=True
+    )
+    lifecycle = models.ForeignKey("Lifecycle", on_delete=models.CASCADE)
```

Comment:
Minor, but IMHO it is better to use the model object itself where possible :thinking:

Reply:
Updated